.github/workflows/forms-backend.yml

```
@@ -0,0 +1,123 @@
+name: Forms Backend
+
+on:
+  push:
+    branches: 
+      - main
+  pull_request:
+
+jobs:
+  lint:
+    name: Linting
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v2
+
+      - name: Setup Python
+        id: python
+        uses: actions/setup-python@v2
+        with:
+          python-version: '3.9'
+      
+      - name: Setup Poetry
+        uses: snok/install-poetry@v1.1.1
+        with:
+          virtualenvs-create: true
+          virtualenvs-in-project: true
+
+      # When same context exists in cache already, restore this environment.
+      - name: Poetry Environment Caching
+        uses: actions/cache@v2
+        id: python_cache
+        with:
+          path: .venv
+          key: "venv-${{ runner.os }}-${{ steps.python.outputs.python-version }}-${{ hashFiles('**/poetry.lock') }}"
+
+      # Only install dependencies when cache didn't hit.
+      - name: Install dependencies
+        if: steps.python_cache.outputs.cache-hit != 'true'
+        run: |
+          poetry install
+
+      # This allows GitHub to detect errors and relay them to PR Files tab
+      - name: Add flake8 matchers
+        run: echo "::add-matcher::./.github/flake8-matchers.json"
+
+      - name: Run flake8
+        run: "poetry run flake8"
+
+      # We don't want that this catch any other errors than flake8 ones.
+      - name: Remove flake8 matchers
+        if: ${{ always() }}
+        run: echo "::remove-matcher owner=flake8-error::" && echo "::remove-matcher owner=flake8-warning::"
+
+  build:
+    name: Build & Push
+    runs-on: ubuntu-latest
+
+    needs: [lint]
+    if: github.ref == 'refs/heads/main'
```

If a PR is made from a fork's "main" branch, the build & push job will also run in your version of the `if` condition, as the `ref` will still be `refs/heads/main`. To prevent that, you should check for the event type:

```suggestion
    if: github.ref == 'refs/heads/main' && github.event_name == 'push'
```